spec/unit/file_system_spec.rb

```diff
@@ -446,6 +446,26 @@ def increment_counter_in_multiple_processes(file, num_procs, options)
       expect(Puppet::FileSystem.readlink(symlink)).to eq(missing_file.to_s)
     end
 
+    it "should be able to unlink a dangling symlink pointed at a file" do
+      symlink = tmpfile("somefile_link")
+      Puppet::FileSystem.symlink(file, symlink)
+      ::File.delete(file)
+      Puppet::FileSystem.unlink(symlink)
+
+      expect(Puppet::FileSystem.exist?(file)).to be_falsey
```

Comment:
minor nit, for predicate methods you'll get better error messages if you do something like:

```
expect(Puppet::FileSystem).to_not be_exist(file)
```